### PR TITLE
Hardening

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 1]
```diff
@@ -27,13 +27,20 @@
 import com.intellij.openapi.fileEditor.FileDocumentManager;
 import com.intellij.openapi.fileEditor.FileEditorManager;
 import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.openapi.project.ProjectUtil;
+import com.intellij.openapi.roots.ContentEntry;
 import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.roots.SourceFolder;
+import com.intellij.openapi.roots.ui.configuration.ModuleConfigurationState;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: MINOR Remove this unused import 'com.intellij.openapi.roots.ui.configuration.ModuleConfigurationState'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 2]
```diff
@@ -108,12 +115,40 @@ public void run() {
...
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if(!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    if(file.getFileType().getDefaultExtension().equalsIgnoreCase("java")) {
+      if(!isSource(file, module)) {
+        SonarLintConsole.get(module.getProject()).debug("Not automatically analysing java file outside source folder: " + file.getName());
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: BLOCKER NullPointerException might be thrown as 'module' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 3]
```diff
@@ -108,12 +115,40 @@ public void run() {
     return false;
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if(!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    if(file.getFileType().getDefaultExtension().equalsIgnoreCase("java")) {
+      if(!isSource(file, module)) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: MAJOR Merge this if statement with the enclosing one. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1066)

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 4]
```diff
@@ -108,12 +115,40 @@ public void run() {
     return false;
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if(!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    if(file.getFileType().getDefaultExtension().equalsIgnoreCase("java")) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: MAJOR Move the "java" string literal on the left side of this string comparison. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1132)
BLOCKER NullPointerException might be thrown as 'file' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 5]
```diff
@@ -27,13 +27,20 @@
 import com.intellij.openapi.fileEditor.FileDocumentManager;
 import com.intellij.openapi.fileEditor.FileEditorManager;
 import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.openapi.project.ProjectUtil;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: MINOR Remove this unused import 'com.intellij.openapi.project.ProjectUtil'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 6]
```diff
@@ -27,13 +27,20 @@
 import com.intellij.openapi.fileEditor.FileDocumentManager;
 import com.intellij.openapi.fileEditor.FileEditorManager;
 import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: MINOR Remove this unused import 'com.intellij.openapi.module.ModuleUtil'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues:
* BLOCKER 2 blocker
* MAJOR 2 major
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
LGTM

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 7]
```diff
@@ -108,12 +112,40 @@ public void run() {
...
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if (!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    // file and module not null here
+    if ("java".equalsIgnoreCase(file.getFileType().getDefaultExtension()) && !isSource(file, module)) {
+      SonarLintConsole.get(module.getProject()).debug("Not automatically analysing java file outside source folder: " + file.getName());
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: BLOCKER NullPointerException might be thrown as 'module' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java [hunk 8]
```diff
@@ -108,12 +112,40 @@ public void run() {
     return false;
   }
 
+  public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    if (!shouldAnalyze(file, module)) {
+      return false;
+    }
+
+    // file and module not null here
+    if ("java".equalsIgnoreCase(file.getFileType().getDefaultExtension()) && !isSource(file, module)) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java]: BLOCKER NullPointerException might be thrown as 'file' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 2 blocker

Watch the comments in this conversation to review them.

**Pull request merged**